javagen/src/main/java/com/azure/autorest/template/example/ModelExampleWriter.java

```diff
@@ -298,6 +307,16 @@ public String accept(ExampleNode node) {
                             initAtSetters.add(String.format(".%1$s(%2$s)", modelProperty.getSetterName(), this.accept(childNode)));
                         }
                     }
+                    for (int i = 0; i < properties.size(); ++i) {
+                        String ctorParameterValue = initAtCtors.get(i);
+                        if ("".equals(ctorParameterValue)) { // not present, due to missing required property
+                            if (properties.get(i).getClientType() instanceof PrimitiveType) {
+                                initAtCtors.set(i, properties.get(i).getClientType().defaultValueExpression());
+                            } else {
+                                initAtCtors.set(i, "null");
+                            }
+                        }
+                    }
```

Comment:
Should this logic be on line 304, when setting the ctor?
https://github.com/Azure/autorest.java/pull/2921/files#diff-5ad12a8cf60cd3edf947aae6e321e17758b17fedb00713c2cdbc29a1e1087bf9R304

setting it, then getting it and compare to `""` is strange.

Reply:
missing node won't reach L304, this is the cause of the bug.. `""` is set during initialization on L299

Maybe we could reverse the iteration, iterating model properties instead of child nodes. This would guarantee each property gets processed.

Reply:
Can we init it as `null`? That at least would make the missing of value apparent.

Reply:
sure